hyp3_autorift/process.py

```diff
@@ -261,6 +261,7 @@ def apply_wallis_nodata_fill_filter(array: np.ndarray, nodata: int) -> Tuple[np.
 
     array, _ = prepare_array_for_filtering(array, nodata)
     filtered, zero_mask = _wallis_filter_fill(array, filter_width=5, std_cutoff=0.25)
+    filtered[zero_mask] = 0
```

Comment:
This might actually be the only change that is needed...

Reply:
Forrest confirmed that it's all that is needed, but the other changes do make the written GeoTIFF no data representation better. 

The zero mask might be better written as a zero mask in the image GeoTIFF instead of a separate file (or by setting the zero mask to `np.nan`) but I'm going to save that for later since it's really hard to predict how autoRIFT will act downstream.